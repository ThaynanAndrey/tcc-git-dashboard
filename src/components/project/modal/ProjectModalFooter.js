import React from 'react';

/**
 * Stateless component with the function to presenting the the footer of New Project Modal.
 * 
 * @author Thaynan Nunes
 */
const ProjectModalFooter = ({ closeModal, createProject }) => {
    return (
        <div className="modal-footer">
            <div className="modal-close waves-effect waves-green btn-flat"
                onClick={closeModal}>
                Cancelar
            </div>
            <div className="modal-close waves-effect waves-green btn-flat"
                onClick={createProject}>
                Adicionar
            </div>
        </div>
    );
}

export default ProjectModalFooter;