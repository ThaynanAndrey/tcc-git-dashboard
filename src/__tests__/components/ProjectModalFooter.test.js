import React from 'react';
import { mount, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import ProjectModalFooter from '../../components/project/modal/ProjectModalFooter';

describe('Test ProjectModalFooter', () => {
  it('renders correctly a component to show ProjectModalFooter', () => {
    const tree = mount(
        <ProjectModalFooter />
    );

    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders correctly a component to show ProjectModalFooter with correct props', () => {
    const [closeModal, createProject] = new Array(2).fill(jest.fn());

    const tree = mount(
        <ProjectModalFooter
            closeModal={closeModal}
            createProject={createProject}
        />
    );

    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.prop('closeModal')).toEqual(closeModal);
    expect(tree.prop('createProject')).toEqual(createProject);
  });

  it('renders correctly a component to show ProjectModalFooter with correct classes', () => {
    const [closeModal, createProject] = new Array(2).fill(jest.fn());

    const tree = mount(
        <ProjectModalFooter
            closeModal={closeModal}
            createProject={createProject}
        />
    );

    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.find('.modal-footer').exists()).toBeTruthy();
    expect(tree.find('.modal-close.waves-effect.waves-green.btn-flat').exists()).toBeTruthy();
  });
})