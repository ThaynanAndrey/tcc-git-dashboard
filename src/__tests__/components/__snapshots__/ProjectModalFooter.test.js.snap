// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProjectModalFooter renders correctly a component to show ProjectModalFooter 1`] = `
<ProjectModalFooter>
  <div
    className="modal-footer"
  >
    <div
      className="modal-close waves-effect waves-green btn-flat"
    >
      Cancelar
    </div>
    <div
      className="modal-close waves-effect waves-green btn-flat"
    >
      Adicionar
    </div>
  </div>
</ProjectModalFooter>
`;

exports[`Test ProjectModalFooter renders correctly a component to show ProjectModalFooter with correct classes 1`] = `
<ProjectModalFooter
  closeModal={[MockFunction]}
  createProject={[MockFunction]}
>
  <div
    className="modal-footer"
  >
    <div
      className="modal-close waves-effect waves-green btn-flat"
      onClick={[MockFunction]}
    >
      Cancelar
    </div>
    <div
      className="modal-close waves-effect waves-green btn-flat"
      onClick={[MockFunction]}
    >
      Adicionar
    </div>
  </div>
</ProjectModalFooter>
`;

exports[`Test ProjectModalFooter renders correctly a component to show ProjectModalFooter with correct props 1`] = `
<ProjectModalFooter
  closeModal={[MockFunction]}
  createProject={[MockFunction]}
>
  <div
    className="modal-footer"
  >
    <div
      className="modal-close waves-effect waves-green btn-flat"
      onClick={[MockFunction]}
    >
      Cancelar
    </div>
    <div
      className="modal-close waves-effect waves-green btn-flat"
      onClick={[MockFunction]}
    >
      Adicionar
    </div>
  </div>
</ProjectModalFooter>
`;
